### PR TITLE
Initial Flow-based component editing, publishing flows, decows api client

### PR DESCRIPTION
**Pull request by dabbott**
Mega PR containing too many features and too many different styles.

Most features are hidden behind a settings flag.

### CODE REVIEW

File: web/src/scripts/components/forms/FormHeader.jsx [hunk 1]
```diff
@@ -17,50 +17,61 @@
...
+    minWidth: 0,
+  },
+  button: {
+    fontSize: 18,
+    fontWeight: 300,
+  }
 }
 
-const FormHeader = ({label, inset, labelWidth, disabled}) => {
+export default class extends Component {
```
Comment (alexzherdev) [inline web/src/scripts/components/forms/FormHeader.jsx]: Just curious, why the change from a SFC to a class component?

File: web/package.json [hunk 2]
```diff
@@ -24,23 +23,28 @@
     "gulp-util": "^3.0.4",
     "json-loader": "^0.5.4",
     "lodash": "^4.0.0",
+    "pure-render-decorator": "^1.1.1",
```
Comment (alexzherdev) [inline web/package.json]: Are there any plans to use React.PureComponent? That would require v15.3.0 though.

File: web/src/scripts/components/publishing/PublishingMetadata.jsx [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+    const {component} = this.state
+    const {props: componentProps = [], dependencies, imports} = component
+
+    const rowDimensions = {
+      labelWidth: width - INPUT_WIDTH - 10,
+      inset: INSET_WIDTH,
+      inputWidth: INPUT_WIDTH,
+    }
+
+    return (
```
Comment (gdowens) [inline web/src/scripts/components/publishing/PublishingMetadata.jsx]: Yo, this render function is madness.

**Comment by gdowens** [comment]
New style is looking so readable, yeh? This is a really solid start.

Only concern: if we don't break up the one or two massive components in there they will continue to gather entropy. I think it's a good idea to spend some minimal effort chunking it into separate components before merging.

Also noticed some code bits had been commented out, we should remove anything that won't be used in the next PR.

File: web/src/scripts/components/forms/FormHeader.jsx (hunk 1 above)
Comment (dabbott) [inline web/src/scripts/components/forms/FormHeader.jsx]: I was using functional components for a while, but:
1. they are less performant, as they can't have a `shouldComponentUpdate`
2. they are inconsistent with the class format

I'm going to use all class components from now on for consistency.

This PR is a bit lazy since I'm short on time, and I picked up many changes from a previous branch. So there are a lot of mixed styles.

File: web/package.json (hunk 2 above)
Comment (dabbott) [inline web/package.json]: Seems like a better idea. 15.3.0 breaks react-virtualized though... so probably wait till that gets fixed.

I'm fine with getting rid of pure-render-decorator though. This is copied over from older code.

File: web/src/scripts/components/publishing/PublishingMetadata.jsx (hunk 3 above)
Comment (dabbott) [inline web/src/scripts/components/publishing/PublishingMetadata.jsx]: Will refactor after I get designs from Roy.

I can redact the component from the PR if you like. You just won't be able to test publishing. But it doesn't work that well yet anyway, so maybe that's the right move.

**Comment by dabbott** [comment]
Left comments inline. I will gauge LoE on removing massive components from PR. If high, will PR next week instead.
